sensors/baro/ms5611.c

```diff
@@ -0,0 +1,257 @@
+	uint32_t D1 = 0, D2 = 0;
+	int32_t dT;
+	int64_t off, sens;
+
+	/* Initialize temperature with too big delay for certain response */
+	while (ms5611_measure(ctx, (CMD_CVRT_TEMP | CVRT_OSR_4096), OSR_4096_SLEEP * 2, &D2) < 0)
+		;
+
+	/* Translate temperature */
+	dT = (int32_t)D2 - (((int32_t)ctx->promC[5]) << 8);           /* dT = D2 - C5 * 2^8 */
+	temp = 2000 + ((int64_t)dT * ((int64_t)ctx->promC[6]) >> 23); /* TEMP = 2000 + dT * C6 / 2^23 */
+
+	while (1) {
+		/* Read pressure */
+		if (ms5611_measure(ctx, (CMD_CVRT_PRESS | CVRT_OSR_4096), OSR_4096_SLEEP, &D1) < 0) {
+			continue;
+		}
+
+		gettime(&now, NULL);
+
+		/* Read temperature. Does not repeat failed read - fall back to old temperature */
+		if (ms5611_measure(ctx, (CMD_CVRT_TEMP | CVRT_OSR_1024), OSR_1024_SLEEP, &D2) == 0) {
+			/* Translate temperature */
+			dT = (int32_t)D2 - (((int32_t)ctx->promC[5]) << 8);           /* dT = D2 - C5 * 2^8 */
+			temp = 2000 + ((int64_t)dT * ((int64_t)ctx->promC[6]) >> 23); /* TEMP = 2000 + dT * C6 / 2^23 */
+		}
+
+		/* Translate press */
+		off = ((int64_t)ctx->promC[2] << 16) + (((int64_t)dT * (int64_t)ctx->promC[4]) >> 7);  /* OFF = C2 * 2^16 + (C4 * dT) / 2^7 */
+		sens = ((int64_t)ctx->promC[1] << 15) + (((int64_t)dT * (int64_t)ctx->promC[3]) >> 8); /* SENS = C1 * 2^15 + (C4 * dT) / 2^8 */
+		press = ((((int64_t)D1 * sens) >> 21) - off) >> 15;                                    /* P = ((D1 * SENS) / 2^21 - OFF) / 2^15 */
+
+		/* MS5611 operating ranges: temperature from -40C to +80C, and pressure: 45000Pa, 110000 Pa */
+		if (press > 45000 && press < 120000 && temp > -4000 && temp < 8500) {
+			ctx->evtBaro.timestamp = now;
+			ctx->evtBaro.baro.pressure = press;
+			ctx->evtBaro.baro.temp = (temp / 100) + 273;
+
+			sensors_publish(info->id, &ctx->evtBaro);
+		}
+	}
+
+	return;
+}
+
+
+static int ms5611_hwSetup(ms5611_ctx_t *ctx)
+{
+	const uint8_t promAddr[6] = { PROM_ADDR_C1, PROM_ADDR_C2, PROM_ADDR_C3, PROM_ADDR_C4, PROM_ADDR_C5, PROM_ADDR_C6 };
+
+	unsigned int i;
+	uint8_t cmd;
+
+	/* Reset ms5611 sequence */
+	cmd = CMD_RESET;
+	if (sensorsspi_xfer(&ctx->spiCtx, &ctx->spiSS, &cmd, sizeof(cmd), NULL, 0, 0) < 0) {
+		fprintf(stderr, "ms5611: failed to reset device\n");
+		return -1;
+	}
+	usleep(RESET_SLEEP);
+
+	/* PROM reading */
+	for (i = 0; i < sizeof(promAddr); i++) {
+		cmd = CMD_READ_PROM | promAddr[i];
+		/* data received from MS5611 is written to incremented index in ctx to match datasheet (so that C1 corresponds to promC[1] and so on) */
+		if (sensorsspi_xfer(&ctx->spiCtx, &ctx->spiSS, &cmd, sizeof(cmd), &ctx->promC[i + 1], sizeof(ctx->promC[i + 1]), sizeof(cmd)) < 0) {
+			fprintf(stderr, "ms5611: failed read PROM C%i\n", i + 1);
+			return -1;
+		}
+		ctx->promC[i + 1] = (ctx->promC[i + 1] << 8) | (ctx->promC[i + 1] >> 8); /* swap higher and lower bytes */
+	}
+
+	printf("C1: %i C2: %i C3: %i C4: %i C5: %i C6: %i\n", ctx->promC[1], ctx->promC[2], ctx->promC[3], ctx->promC[4], ctx->promC[5], ctx->promC[6]);
```

Comment:
I suppose that it is a debugging msg. It will not provide any valuable information for user in runtime.